install_scripts/templates/common/kubernetes.sh

```diff
@@ -990,11 +991,21 @@ apiServer:
     bind-address: $PRIVATE_ADDRESS
     service-node-port-range: "80-60000"
   certSANs:
-  - $PRIVATE_ADDRESS
 EOF
     if [ -n "$PUBLIC_ADDRESS" ]; then
-          cat <<EOF >> /opt/replicated/kubeadm.conf
+        cat <<EOF >> /opt/replicated/kubeadm.conf
   - $PUBLIC_ADDRESS
+EOF
+    fi
+    if [ -n "$LOAD_BALANCER_ADDRESS" ] && [ -n "$LOAD_BALANCER_PORT" ]; then
+        cat <<EOF >> /opt/replicated/kubeadm.conf
+  - $LOAD_BALANCER_ADDRESS
+controlPlaneEndpoint: "$LOAD_BALANCER_ADDRESS:$LOAD_BALANCER_PORT"
+EOF
+    else
+        cat <<EOF >> /opt/replicated/kubeadm.conf
+  - $PRIVATE_ADDRESS
+controlPlaneEndpoint: "$PRIVATE_ADDRESS"
```

Comment:
do you need the port here?

Reply:
i think it will default to the correct port https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go#L75-L76